Telepathy/Server.cs

```diff
@@ -250,6 +243,38 @@ void Listen(int port)
             }
         }
 
+        bool TryListen(int port)
+        {
+            try
+            {
+                // start listener
+                listener = new TcpListener(new IPEndPoint(IPAddress.Any, port));
+                listener.Server.NoDelay = NoDelay;
+                listener.Server.SendTimeout = SendTimeout;
+                listener.Start();
+                Logger.Log("Server: listening port=" + port);
+                return true;
+            }
+            catch (ThreadAbortException exception)
+            {
+                // UnityEditor causes AbortException if thread is still
+                // running when we press Play again next time. that's okay.
+                Logger.Log("Server thread aborted. That's okay. " + exception);
```

Comment:
I can certainly also add that, but the return type of Start() is not for the user. Looking through Mirror, in NetworkServer's Listen function there is some unreachable code meaning that it will never return false. I think telepathy Server's Start return value should be used internally to determine if the server started correctly.

Specifaclly, I think Transport's ServerStart function should return a bool instead of void so that NetworkServer's Listen function can return false is the transport does not start correctly. This will also fix the unreachable code in NetworkManager's StartServer function.

This part of the StartServer function in NetworkManager can NEVER return false, meaning that StartServer() always returns true to the user, and ServerChangeScene will run even if the server did not start correctly.
```
if (!NetworkServer.Listen(maxConnections))
{
    Debug.LogError("StartServer listen failed.");
    return false;
}
```
It seems like a pretty standard thing to have each transport return whether or not it started correctly. You could even add a virtual function StartServerFeedback to Transport to keep all existing transports backwards compatible.
```
StartServerFeedback() { StartServer(); return true; }
```
Making these changes would fix the internal start problems, such as unreachable code, and allow the user to know if the server started correctly which is very important for making sure remote servers kill themselves if they do not start correctly.